pkg/pub_worker/lib/src/bin/pana_wrapper.dart

```diff
@@ -162,50 +165,103 @@ Future<(SdkConfig, SdkConfig)> _detectSdks(Pubspec pubspec) async {
   // Discover installed Dart and Flutter SDKs.
   // This reads sibling folders to the Dart and Flutter SDK.
   // TODO: Install Dart / Flutter SDKs into these folders ondemand in the future!
-  final dartSdks = await InstalledSdk.fromDirectory(
+  final dartSdks = await InstalledSdk.scanDirectory(
     kind: 'dart',
     path: Directory(
       Platform.environment['DART_SDK'] ??
           Directory(Platform.resolvedExecutable).parent.parent.path,
     ).parent,
   );
-  final flutterSdks = await InstalledSdk.fromDirectory(
+  final flutterSdks = await InstalledSdk.scanDirectory(
     kind: 'flutter',
     path: Directory(Platform.environment['FLUTTER_ROOT'] ?? '').parent,
   );
```

Comment:
Why do we scan here? are we not only interested in the stable version?

And are we not interested in NOT being affected by previous pana invocations?

Reply:
Shouldn't this be:
 * (A) Find the stable SDK that was shipped in the current docker image
 * If (A) works, use it!
 * (B) List available prerelease SDKs (beta and dev releases)
 * (C) Find the oldest SDK from (B) that satisfies the SDK-constraints for the package
 * Check if the SDK found in (C) has already been downloaded, if so use it
 * Download the SDK found in (C), install it and use it.

Reply:
This way, multiple pre-releases of the same SDK doesn't affect eachother.

I'm a little uncertain of (C). Choosing the minimum SDK that satisfies the SDK constraint makes some sense. That way the SDK won't change when the package is re-analyzed.

On the flip side, analyzing with the latest SDK pre-release could make sense too.

------------

oh, I'm missing logic with preferable a new stable SDK if one is available, and the current one in the image doesn't satisfy the SDK constraint.

Reply:
So maybe the priority is:

 * (1) SDK from the docker image, if that satisfies constraints
 * (2) Latest stable SDK, if that satisfies constraints
 * (3) Latest beta SDK, if that satisfies constraints
 * (4) Latest dev SDK, if that satisfies constraints

Once an SDK has been identified, check if it's installed, if not install it.

Reply:
Agreed on most parts, adding a TODO and will do a follow-up in later PRs.